interp/interp.go

```diff
@@ -0,0 +1,176 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package interp
+
+import (
+	"fmt"
+	"math"
+	"sort"
+)
+
+// Interpolator1D represents a 1D interpolator.
+// It interpolates a y(x) function based on some data over a range [begin(), end()].
+// Both begin() and end() can be infinite, but begin() must be lower than end().
+type Interpolator1D interface {
+	// Returns the lowest allowed argument for evaluate().
+	begin() float64
+
+	// Returns the highest allowed argument for evaluate().
+	end() float64
+
+	// Evaluates interpolated sequence at x in [begin(), end()].
+	// Panics if the argument is outside this range.
+	eval(float64) float64
+}
+
+// ConstInterpolator1D is a constant 1D interpolator,
+// It is defined over the range [-infinity, infinity].
+type ConstInterpolator1D struct {
+	// Constant Y value.
+	value float64
```

Comment:
if we were to export that field, we could get rid of the `NewConstInterpolator1D` function.